tests/integration/suite/daprd/lock/grpc/errors.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	apiErrors "github.com/dapr/dapr/pkg/api/errors"
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/suite"
+	kitErrors "github.com/dapr/kit/errors"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc"
+	grpcCodes "google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+)
+
+func init() {
+	suite.Register(new(standardizedErrors))
+}
+
+type standardizedErrors struct {
+	daprd *daprd.Daprd
+}
+
+func (e *standardizedErrors) Setup(t *testing.T) []framework.Option {
+	e.daprd = daprd.New(t,
+		daprd.WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: lockstore
+spec:
+  type: lock.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+`))
```

Comment:
The configurations were successful when tested locally, but I'm uncertain if they'll function properly in the CI environment. If they don't, I'm unsure about the correct values to assign to these parameters.

Reply:
Hi @moonorange , do you need a Redis instance, specifically, or just any key-value store? Would the default in-memory store work?

Reply:
Ah, we're working with a lockstore here, not a state store. So basically, we should check if we have a default in-memory lockstore, if we do, it will have been used in integration tests already. If not, we're gonna have to create a pluggable component, as illustrated in the file above.

Reply:
Looks we only have a redis lock for now: https://github.com/dapr/components-contrib/tree/main/lock.
You can simulate one by creating your own pluggable component, and simulating the Redis behaviour.

Reply:
@JoshVanL, I wonder if it would make more sense to create an in-memory lock component in components-contrib? Do you see a scenario where it might be useful to users?

Reply:
@elena-kolevska According to the following document, pluggable components are not currently supported for the lock store. Is there a way to simulate Redis behavior?

- https://docs.dapr.io/developing-applications/develop-components/pluggable-components/develop-pluggable/

Reply:
cc @elena-kolevska

Reply:
@JoshVanL what do you think about the comment above, creating an in-memory lock component?

Reply:
Yup, though it would be my preference to use a file based lock.

Reply:
Thanks Josh!
@moonorange first of all, thank you once again for the great work! The next step before we can merge this PR is getting the integration tests right. As you noted yourself, this is not straightforward because we don't run Redis in the CI for integration tests, so we have to be creative!  

I believe the easiest way would be to create a new file-based lock component in the [components-contrib](https://github.com/dapr/components-contrib) repo. In order to run in the CI without any external dependencies it needs to be file-based (so it can be shared across apps). 

I think this can be a really fun exercise! You would basically have to look at the [redis-based lock component](https://github.com/dapr/components-contrib/blob/main/lock/redis/standalone.go) and duplicate its functionality using files as locks. Once that's done, you can set your local dapr to pull from your updated components-contrib  by modifying `dapr/go.mod` to point to it (scroll down to the line that does a `go replace` for components contrib, uncomment it and change it to the path to your local components-contrib repo).
And then, you can set the test component lock store to your new file-based lockstore.

Do you feel up for the challenge? :) We're always here to help out if you get stuck!

Reply:
@elena-kolevska 
Thank you for the pointer. I'm totally down for this challenge!

I'll start by checking out how the Redis-based lock component works and then figure out how to replicate it using files. If I hit any bumps, I'll definitely reach out for a hand.

Reply:
Amazing! I'm happy to hear that.
It would be great if we could get this in the next release (1.14). Code freeze for it is the 11th of June.